enhancements/namespace-auto-pruning.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Namespace Auto-pruning Policies
+authors:
+  - "@bcaton85"
+reviewers:
+  - "@dmage"
+  - "@Sunandadadi"
+approvers:
+  - "@dmage"
+  - "@Sunandadadi"
+creation-date: 2023-08-15
+last-updated: 2023-08-15
+status: provisional
+---
+
+# Namespace Auto-pruning Policies
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+1. Should there be a dry run feature to test the policy?
+1. Should this only be implemented in the new UI and not the old UI?
+1. Can a user configure multiple policies per namespace?
+1. Can a user specify tags or repositories by pattern?
+1. Do we want the ability to have tiered priority for auto-prune execution?
+
```

Comment:
How do these autopruning policies work in a Proxy Organization? Proxy Organizations already have a form of autopruning in effect if storage quota is enabled on that Org. If we allow these autopruning policies on a Proxy Organization with quota based autopruning in effect, could they conflict with each other?

Reply:
Yes, that's a great question. They shouldn't conflict with each other since it's essentially the same as two tags being manually deleted at the same time. This is something we'll have to keep in mind during implementation.

Reply:
As @BillDett mentioned, proxy organizations already have a form of auto-pruning in place. Should we eventually migrate this proxy specific auto-pruning into one of the options users can enable on a namespace level? The proxy specific auto-pruning is very similar to the age policy.